### PR TITLE
Vcb-86 refactor binary types

### PR DESCRIPTION
**Pull request by m-dot-wigger**
merge conflicts should be solved

### CODE REVIEW

**Comment by m-dot-wigger** [comment]
@clooudch Has a circular dependency at the binary types any idea how to solve it?

**Comment by m-dot-wigger** [comment]
> > Has a circular dependency at the binary types any idea how to solve it?
> 
> You can use interfaces.

solved more or less nicely

File: src/types/binary/binaryType.ts [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+  /**
+   * Adds the specified number to the byte and returns the result as a new byte. In case the
+   * result exceeds the `Byte.MAX_VALUE` then it will overflow.
+   *
+   * @param value the number to be added to the byte
+   * @returns the new byte with the value added
+   */
+  public add(value: Byte | number): Byte {
+    return new Byte(
+      limitValuesToBitCount(this.addToNumber(value), this.numberOfBitsForType)
```
Comment (deflox) [inline src/types/binary/binaryType.ts:275]: Why do we not use the abstract max value and make an & operation with it? To me this seems to be a bit of an overhead. But I guess this is more personal preference.

Edit: I just saw that you do some additional checks like safeInteger etc. I guess then its fine.

**Review by deflox — APPROVED** [review]
The rest looks good to me. Tests are all successful for me on my machine.

File: src/types/binary/binaryType.ts (hunk 1 above)
Comment (m-dot-wigger) [reply src/types/binary/binaryType.ts:275]: you are right, since with our 3 binary types we are always within safe integer range for add operation

Comment (m-dot-wigger) [reply src/types/binary/binaryType.ts]: keep the unused methods `limitValuesToBitCount, getMaxValueDisplayableWithBits` perhaps can be used for something else in future

**Comment by m-dot-wigger** [comment]
@deflox and @clooudch 
Can you please have another look in this PR?

**Review by m-dot-wigger — COMMENTED** [review]
Most files are just reordering of imports. The following files include "real" changes:

- all files in project root like README.md and config files
- src/board/alu.ts
- src/types/binary/binaryType.ts
- src/types/binary/halfword.ts
- src/types/binary/utils.ts
- src/types/binary/word.ts
- src/types/error/index.ts
- test/instruction/opcode.test.ts
- test/instruction/set.test.ts
- test/types/binary/byte.test.ts
- test/types/binary/halfword.test.ts
- test/types/binary/utils.test.ts
- test/types/binary/word.test.ts

**Review by deflox — APPROVED** [review]
Looks good. The only thing which seems a bit odd to me is why the "Byte" type also included in the "binaryType.ts" instead of its own file like before.

**Comment by m-dot-wigger** [comment]
> Looks good. The only thing which seems a bit odd to me is why the "Byte" type also included in the "binaryType.ts" instead of its own file like before.

That's to "solve" the circular dependency

**Pull request merged**